Sources/NostrSDK/Events/Calendars/CalendarNostrEvent.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  CalendarNostrEvent.swift
+//
+//
+//  Created by Terry Yiu on 12/1/23.
+//
+
+import Foundation
+
+/// A calendar is a collection of calendar events, represented as a custom replaceable list event.
+/// A user can have multiple calendars.
+/// One may create a calendar to segment calendar events for specific purposes. e.g., personal, work, travel, meetups, and conferences.
+///
+/// See [NIP-52 - Calendar](https://github.com/nostr-protocol/nips/blob/master/52.md#calendar).
+public final class CalendarNostrEvent: NostrEvent, IdentifierTagInterpreting, TitleTagInterpreting {
+    public required init(from decoder: Decoder) throws {
+        try super.init(from: decoder)
+    }
+
+    @available(*, unavailable, message: "This initializer is unavailable for this class.")
+    override init(kind: EventKind, content: String, tags: [Tag] = [], createdAt: Int64 = Int64(Date.now.timeIntervalSince1970), signedBy keypair: Keypair) throws {
+        try super.init(kind: kind, content: content, tags: tags, createdAt: createdAt, signedBy: keypair)
+    }
+
+    public init(content: String, tags: [Tag] = [], createdAt: Int64 = Int64(Date.now.timeIntervalSince1970), signedBy keypair: Keypair) throws {
+        try super.init(kind: .calendar, content: content, tags: tags, createdAt: createdAt, signedBy: keypair)
+    }
+
+    /// The event coordinates of the calendar events that belong to this calendar.
+    public var calendarEventsCoordinates: [EventCoordinates] {
```

Comment:
Naming here is tough. An `EventCoordinates` is a singular thing, though it sounds plural. The word "calendar" here feels redundant. What if this were `allEventsCoordinates`?

Reply:
I think I wanted to make it clear that these are only the event coordinates that are calendar events. As in other kinds are filtered out. Not sure of a better name though.

Reply:
Maybe just `calendarEvents` or `calendarList`?

Reply:
Or maybe `calendarEventCoordinateList`.

Reply:
Changed to `calendarEventCoordinateList`. Let me know what you think.

Reply:
In general I don't want "array" or "list" in variable names; they should just be a plural noun. In this case, the word "coordinates" complicates things, so I'm okay with it for this one.